### PR TITLE
fix(plasma-temple): improve type-coverage

### PR DESCRIPTION
**Pull request by Yeti-or**
:baby_chick: Download canary assets:

[colors--canary.239.3547886503.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.239.3547886503.xml)  
[color_sbermarket_ios-swift--canary.239.3547886503.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.239.3547886503.swift)  
[color_sbermarket_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.239.3547886503.kt)  
[color_sbermarket_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.239.3547886503.ts)  
[color_sberprime_ios-swift--canary.239.3547886503.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.239.3547886503.swift)  
[color_sberprime_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.239.3547886503.kt)  
[color_sberprime_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.239.3547886503.ts)  
[PlasmaTokensColor--canary.239.3547886503.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.239.3547886503.swift)  
[typo_mage_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.239.3547886503.kt)  
[typo_mage_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.239.3547886503.ts)  
[typo_plasma_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.239.3547886503.kt)  
[typo_plasma_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.239.3547886503.ts)  
[typo_ruler_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.239.3547886503.kt)  
[typo_ruler_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.239.3547886503.ts)  
[typo_sage_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.239.3547886503.kt)  
[typo_sage_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.239.3547886503.ts)  
[typo_sbermarket_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.239.3547886503.kt)  
[typo_sbermarket_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.239.3547886503.ts)  
[typo_soulmate_kotlin--canary.239.3547886503.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.239.3547886503.kt)  
[typo_soulmate_react-native--canary.239.3547886503.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.239.3547886503.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-temple@1.114.3-canary.239.3547886503.0
  # or 
  yarn add @salutejs/plasma-temple@1.114.3-canary.239.3547886503.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-239/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-239/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-239/
  ui storybook: http://plasma.sberdevices.ru/pr-239/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-239/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-239/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-239/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-239/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/temple-storybook/

  </details>

File: packages/plasma-temple/src/types.ts [hunk 1]
```diff
@@ -73,13 +74,15 @@ export interface AssistantAppStateItem {
     };
 }
 
-export interface AssistantAppState {
+export interface AssistantAppState extends _AssistantAppState {
+    /*
     // eslint-disable-next-line camelcase
     item_selector: {
         items: Array<AssistantAppStateItem>;
     };
```
Comment (Yakutoc) [inline packages/plasma-temple/src/types.ts:82]: @Yeti-or Почему закомментировано с 78 по 82 и 85 строчки?

Comment (Yeti-or) [reply packages/plasma-temple/src/types.ts:82]: хз, уберу

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-temple/src/components/AssistantProvider/AssistantProvider.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ export function AssistantProvider<A extends AssistantSmartAppData = AssistantSma
         onData: handleData,
     });
 
-    const value: AssistantContextType = React.useMemo(() => ({ assistant, setAssistantState, state }), [
+    const value = React.useMemo<AssistantContextType>(() => ({ assistant, setAssistantState, state }), [
```
Comment (LamaEats) [inline packages/plasma-temple/src/components/AssistantProvider/AssistantProvider.tsx:59]: а есть ли разница?

File: packages/plasma-temple/src/components/Form/components/DatePicker.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ type DatePickerProps = FieldComponentProps<
 
 export const DatePicker = withWrapField<Date, DatePickerProps>(
     // eslint-disable-next-line prefer-arrow-callback
-    function DatePicker(props) {
+    function DatePicker(props: DatePickerProps) {
```
Comment (LamaEats) [inline packages/plasma-temple/src/components/Form/components/DatePicker.tsx:21]: тут же ТС сам нормально выводил тип пропсов

File: packages/plasma-temple/src/components/Form/components/Form.tsx [hunk 4]
```diff
@@ -47,7 +47,7 @@ export function Form<D, K extends keyof D = keyof D>(props: FormProps<D, K>): Re
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [sequence, currentField, formState, onChangeValueField, onSubmit]);
 
-    const onChangeHandler = React.useCallback(
+    const onChangeHandler = React.useCallback<ChangeEventHandler>(
```
Comment (LamaEats) [inline packages/plasma-temple/src/components/Form/components/Form.tsx:50]: тут у value -> event.target.value

File: packages/plasma-temple/src/components/Product/ProductDescription/ProductDescription@mobile.tsx [hunk 5]
```diff
@@ -131,7 +131,7 @@ const getLines = (str: string, maxLengthRow: number) => {
 
 const padLines = (lines: string[], length: number) => {
     if (lines.length < length) {
-        return [...Array(length - lines.length), ...lines];
+        return [...Array<undefined>(length - lines.length), ...lines];
```
Comment (LamaEats) [inline packages/plasma-temple/src/components/Product/ProductDescription/ProductDescription@mobile.tsx:134]: ну там вроде не совсем `undefined` скорее `void` или тс тут ругается?

File: packages/plasma-temple/src/pages/ConfirmOrderPage/components/ConfirmOrderCard/ConfirmOrderCard.tsx [hunk 6]
```diff
@@ -29,7 +29,7 @@ const StyledCaption = styled(Caption)`
     color: ${secondary};
 `;
 
-const mapDeviceToContent: Record<DeviceFamily, React.FC> = {
+const mapDeviceToContent: Record<DeviceFamily, React.FC<{ children?: ReactNode }>> = {
```
Comment (LamaEats) [inline packages/plasma-temple/src/pages/ConfirmOrderPage/components/ConfirmOrderCard/ConfirmOrderCard.tsx:32]: а чего не `PropsWithChildren<{}>`?

File: packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts [hunk 7]
```diff
@@ -1,13 +1,8 @@
...
 
-import { GetStyledComponentProps } from '../../../types';
-
 export type CarouselProps = Omit<BaseCarouselProps, 'detectActive' | 'detectThreshold'>;
 
-type StyledComponentProps = GetStyledComponentProps<typeof Carousel>;
-type StyledCarousel = StyledComponent<typeof Carousel, any, StyledComponentProps, never>;
-
-export const CommonCarousel: StyledCarousel = styled(Carousel)`
+export const CommonCarousel: typeof Carousel = styled(Carousel)`
```
Comment (LamaEats) [inline packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: а чем выведенный тип плох?

**Review by LamaEats — APPROVED** [review]

File: packages/plasma-temple/src/components/Form/components/DatePicker.tsx (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/Form/components/DatePicker.tsx:21]: Кажется вообще можно убрать

File: packages/plasma-temple/src/components/Form/components/Form.tsx (hunk 4 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/Form/components/Form.tsx:50]: не понял что ты имеешь ввиду

File: packages/plasma-temple/src/components/Product/ProductDescription/ProductDescription@mobile.tsx (hunk 5 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/Product/ProductDescription/ProductDescription@mobile.tsx:134]: слушай хороший вопрос, с одной стороны кажется что да, но я не уверен что `<empty slot>` в массиве это тоже самое что `void`; типа void это когда нам важно чтобы ф-ия ничего не возвращала; а вот пустые слоты 
кажется другое дело

ну и если сделать `[...Array(2)]` то будет `[undefined, undefined]` 
а если указать `[...Array<void>(2)]` то ts будет думать что там массив`void[]`
что не верно, поэтому undefined

File: packages/plasma-temple/src/pages/ConfirmOrderPage/components/ConfirmOrderCard/ConfirmOrderCard.tsx (hunk 6 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/pages/ConfirmOrderPage/components/ConfirmOrderCard/ConfirmOrderCard.tsx:32]: поменял

File: packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts (hunk 7 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: тем что при  `npm ci` оно работает
а при `npx lerna bootstap` нихера

```sh
Error: src/pages/GalleryPage/components/Carousel.ts(6,14): error TS2742: The inferred type of 'CommonCarousel' cannot be named without a reference to '@salutejs/plasma-ui/node_modules/@salutejs/plasma-core'. This is likely not portable. A type annotation is necessary.
Error: src/pages/GalleryPage/components/Carousel.ts(6,14): error TS2742: The inferred type of 'CommonCarousel' cannot be named without a reference to '@salutejs/plasma-ui/node_modules/@salutejs/plasma-core/components/Carousel/types'. This is likely not portable. A type annotation is necessary.
```

Я не понял что предлаегается тут для монорепы: https://typescript.tv/errors/#TS2742
ибо первый раз такая проблема

возможно это от того что в tsconfis.json в plasma-temple написано:

```
        "outDir": "dist",
        "rootDir": "src",
        "baseUrl": ".",
        "paths": {
            "@salutejs/plasma-ui": ["node_modules/@salutejs/plasma-ui"]
        },
        "preserveSymlinks": true

```
но я не знаю на самом деле зачем там путь до plasma-ui поэтому залатал вот таким образом

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-239/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-239/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-239/
  ui storybook: http://plasma.sberdevices.ru/pr-239/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-239/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-239/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-239/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-239/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/temple-storybook/

  </details>

File: packages/plasma-temple/src/components/AssistantProvider/AssistantProvider.tsx (hunk 2 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/AssistantProvider/AssistantProvider.tsx]: кажется можно вообще аннтацию убрать и он будет выводить типы

File: packages/plasma-temple/src/components/Form/components/DatePicker.tsx (hunk 3 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/Form/components/DatePicker.tsx:21]: я хрен знает почему, но нет
Screenshot 2022-11-24 at 22 21 12

Comment (LamaEats) [reply packages/plasma-temple/src/components/Form/components/DatePicker.tsx:21]: О_о

File: packages/plasma-temple/src/components/Form/components/Form.tsx (hunk 4 above)
Comment (LamaEats) [reply packages/plasma-temple/src/components/Form/components/Form.tsx:50]: `ChangeEventHandler` функция, у которой аргумент `React.ChangeEvent`, а тут функция получает только `value` элемента, с которого эвент произошел

File: packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts (hunk 7 above)
Comment (LamaEats) [reply packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: не, я про тот который был, кастомный для вытаскивание пропсов компонентов

File: packages/plasma-temple/src/components/Form/components/DatePicker.tsx (hunk 3 above)
Comment (LamaEats) [reply packages/plasma-temple/src/components/Form/components/DatePicker.tsx:21]: когда-нибудь надо там типы поправить, они не работают

File: packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts (hunk 7 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: а он мне  any возвращал  ¯\_(ツ)_/¯

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-239/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-239/

  </details>

File: packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts (hunk 7 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: А нет, не он,  а сами пропсы

Screenshot 2022-11-25 at 14 56 03

Comment (Yeti-or) [reply packages/plasma-temple/src/pages/GalleryPage/components/Carousel.ts:6]: Вот так тебе больше нравится? 

```ts
import styled, { StyledComponent } from 'styled-components';
import { Carousel, CarouselProps as BaseCarouselProps } from '@salutejs/plasma-ui';

export type CarouselProps = Omit<BaseCarouselProps, 'detectActive' | 'detectThreshold'>;

type StyledCarousel = StyledComponent<typeof Carousel, any, CarouselProps, never>;

export const CommonCarousel: StyledCarousel= styled(Carousel)`
    padding-right: var(--plasma-grid-margin);
`;

```

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-239/
  ui storybook: http://plasma.sberdevices.ru/pr-239/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-239/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-239/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-239/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-239/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/temple-storybook/

  </details>

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-239/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-239/

  </details>

File: packages/plasma-temple/src/components/Form/components/Form.tsx (hunk 4 above)
Comment (Yeti-or) [reply packages/plasma-temple/src/components/Form/components/Form.tsx]: поправил

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-239/
  ui storybook: http://plasma.sberdevices.ru/pr-239/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-239/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-239/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-239/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-239/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-239/temple-storybook/

  </details>

**Pull request merged**